tests/test_metrics.py

```diff
@@ -190,6 +195,114 @@ def testValidEmpty(self):
+        self.endTask.run()
+
+    def testSingleTask(self):
+        task = PipelineTimingMetricTask(config=self._makeConfig(nameStart="first", nameEnd="first"))
+
+        altConfig = lsst.verify.tasks.TimingMetricConfig()
+        altConfig.connections.labelName = "first"
+        altConfig.target = "first.run"
+        altConfig.connections.package = "verify"
+        altConfig.connections.metric = "DummyTime"
+        altTask = lsst.verify.tasks.TimingMetricTask(config=altConfig)
+
+        result = task.run(self.startTask.getFullMetadata(), self.startTask.getFullMetadata())
+        oracle = altTask.run(self.startTask.getFullMetadata())
+
+        self.assertEqual(result.measurement.metric_name, self.metric)
+        self.assertAlmostEqual(result.measurement.quantity.to_value(u.s),
+                               oracle.measurement.quantity.to_value(u.s))
+
+    def testTwoTasks(self):
+        firstTask = PipelineTimingMetricTask(config=self._makeConfig(nameStart="first", nameEnd="first"))
+        secondTask = PipelineTimingMetricTask(config=self._makeConfig(nameStart="last", nameEnd="last"))
+
+        result = self.task.run(self.startTask.getFullMetadata(), self.endTask.getFullMetadata())
+        firstResult = firstTask.run(self.startTask.getFullMetadata(), self.startTask.getFullMetadata())
+        secondResult = secondTask.run(self.endTask.getFullMetadata(), self.endTask.getFullMetadata())
+
+        self.assertEqual(result.measurement.metric_name, self.metric)
+        self.assertGreater(result.measurement.quantity, 0.0 * u.s)
+        self.assertGreaterEqual(result.measurement.quantity,
+                                firstResult.measurement.quantity + secondResult.measurement.quantity)
+
+    def testRunDifferentMethodFirst(self):
+        config = self._makeConfig(nameStart="first", nameEnd="last")
+        config.targetStart = "first.doProcess"
+        task = PipelineTimingMetricTask(config=config)
+        try:
+            result = task.run(self.startTask.getFullMetadata(), self.endTask.getFullMetadata())
+        except lsst.pipe.base.NoWorkFound:
+            # Correct behavior
+            pass
+        else:
+            # Alternative correct behavior
+            testUtils.assertValidOutput(task, result)
+            meas = result.measurement
+            self.assertIsNone(meas)
+
+    def testRunDifferentMethodLast(self):
+        config = self._makeConfig(nameStart="first", nameEnd="last")
+        config.targetStart = "last.doProcess"
+        task = PipelineTimingMetricTask(config=config)
+        try:
+            result = task.run(self.startTask.getFullMetadata(), self.endTask.getFullMetadata())
+        except lsst.pipe.base.NoWorkFound:
+            # Correct behavior
+            pass
+        else:
+            # Alternative correct behavior
+            testUtils.assertValidOutput(task, result)
+            meas = result.measurement
+            self.assertIsNone(meas)
+
+    def testBadlyTypedKeys(self):
```

Comment:
The Task itself catches both TypeError and ValueError; this only checks the former, and it's not clear to me why the latter is necessary. Unfortunately, `datetime.fromisoformat` doesn't list what it could raise, that I can find. Catching two exceptions in the test above made me wonder what the examples were to trigger each of them.

Reply:
I'll admit I adapted the code from the existing `TimingMetricTask` and don't remember, but `ValueError` seems a reasonable exception for a string that is _not_ an ISO date.